### PR TITLE
Fix blankposts not showing up in logs

### PR DESCRIPTION
**Pull request by Crystalwarrior**

### CODE REVIEW

**Comment by stonedDiscord** [comment]
i think that was a feature at some point

**Review by oldmud0 — CHANGES_REQUESTED** [review]
Why did you have to write a cursor? Why can't you just call `.size() - 1`

**Comment by Crystalwarrior** [comment]
> Why did you have to write a cursor? Why can't you just call `.size() - 1`

View desync

**Comment by oldmud0** [comment]
> > Why did you have to write a cursor? Why can't you just call `.size() - 1`
> 
> View desync

Wait but you only access one element which is like the "last logged message." Why can't you store that. I'm trying to avoid an issue later on forgetting to increment/decrement something and causing an OOB.

**Comment by Crystalwarrior** [comment]
> > > Why did you have to write a cursor? Why can't you just call `.size() - 1`
> > 
> > View desync
> 
> Wait but you only access one element which is like the "last logged message." I think you should store that value directly instead. I'm trying to avoid an issue later on forgetting to increment/decrement something and causing an OOB.

**Comment by Crystalwarrior** [comment]
Figured out a better method lol

**Review by stonedDiscord — APPROVED** [review]
the counter is gone and there's less lines

File: src/courtroom.cpp [hunk 1]
```diff
@@ -5243,12 +5244,16 @@ void Courtroom::on_showname_enable_clicked()
 void Courtroom::regenerate_ic_chatlog()
 {
   ui_ic_chatlog->clear();
+  last_ic_message = "";
   foreach (chatlogpiece item, ic_chatlog_history) {
-    append_ic_text(item.get_message(),
-                   ui_showname_enable->isChecked() ? item.get_showname()
-                                                   : item.get_name(),
+    QString message = item.get_message();
```
Comment (oldmud0) [inline src/courtroom.cpp:5249]: (We discussed that this is trimmed beforehand, so all blankposts are guaranteed to be empty strings.)

File: src/courtroom.cpp [hunk 2]
```diff
@@ -2153,13 +2153,14 @@ void Courtroom::display_log_chatmessage(QString f_message, int f_char_id, QStrin
...
 
-  // If the chat message isn't a blankpost, or the chatlog history is empty, or its last message isn't a blankpost
-  if (!f_message.isEmpty() ||
-      ic_chatlog_history.isEmpty() || ic_chatlog_history.last().get_message() != "") {
-    // Append the message to the IC chatlogs in client
-    append_ic_text(f_message, f_displayname, "",
-                   f_color);
-  }
+  // If our current message is a blankpost, the chat log isn't empty, the chat log's last message is a blank post, and the blankpost's showname is the same as ours
+  if (f_message.isEmpty() && last_ic_message == f_displayname + ":")
```
Comment (oldmud0) [inline src/courtroom.cpp:2157]: Duplicate comment with different conditions. Not questioning anything

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**